internal/testrunner/testrunner.go

```diff
@@ -66,6 +68,23 @@ type TestRunner interface {
 	TearDown(context.Context) error
 }
 
+type TesterFactory func(TestFolder) (Tester, error)
+
+// TestRunner is the interface test runners that require a global initialization must implement.
+type TestRunner interface {
+	// Type returns the test runner's type.
+	Type() TestType
+
+	// SetupRunner prepares global resources required by the test runner.
+	SetupRunner(context.Context) error
+
+	GetTests(context.Context) ([]TestFolder, error)
```

Comment:
Umm, I would expect this to directly return test[er]s.
```suggestion
	GetTests(context.Context) ([]Tester, error)
```

Reply:
Update method to return a Tester instead of TestFolder.
Each tester returned must have all the configuration / parameters set.